### ISSUE TITLE
Only Display Region Name

### ISSUE THREAD

**Issue by TurtleP** [maptz/maptz.vscode.extensions.customfolding#117]
Hi,

Something that I've been trying to figure out with this extension is making the fold only display a name given to it. I've modified my `settings.json`, although I'm not entirely certain what I'm doing is correct, even after looking at the examples.

```json
  "maptz.regionfolder": {
    "[cpp]": {
      "foldEnd": "[NAME]",
      "foldEndRegex": "/\\*[\\s]*#endregion",
      "foldStart": "[NAME]",
      "foldStartRegex": "^[\\s]*/\\*[\\s]*#region[\\s]*(.*)[\\s]*\\*/[\\s]*$"
    }
  }
```

Regardless, if this isn't a feature that's in the extension, I think  it would be nice to include. I personally don't care to see how my comments are laid out for a region, given I *know* how the comments look syntactically and would prefer to only see what the name of a region is. For example:

 ```cpp
/* #region "Something" */
void test()
{
}
/* #endregion */
```
would fold into just the text `Something`, but styled as a comment.

**Comment by maptz** [comment]
Yes. I agree, and this has been requested before. Last time I looked, it's wasn't possible given the VS Extension API. Perhaps that has changed in the intervening months. If I get a chance, I'll take a look.

**Comment by TurtleP** [comment]
Ah, I did look for an issue about this, but didn't find anything, which is why I opened this. Thanks for looking into it!

**Comment by Janfel** [comment]
> Last time I looked, it's wasn't possible given the VS Extension API. Perhaps that has changed in the intervening months.

I don’t know much about the VSCode API, but [`mohammadbaqer.better-folding`](https://marketplace.visualstudio.com/items?itemName=MohammadBaqer.better-folding) provides something like a non-configurable version of this extension that has this feature. It seems to use Decorations to hide the range objects. I have linked the code I believe does this below.

https://github.com/mtbaqer/vscode-better-folding/blob/7db960eef7a180f7d3775c5493a946be0802b988/src/providers/regionRangesProvider.ts#L26
https://github.com/mtbaqer/vscode-better-folding/blob/7db960eef7a180f7d3775c5493a946be0802b988/src/decorators/betterFoldingDecorator.ts#L32

Maybe you can talk to the author of `mohammadbaqer.better-folding` to collaborate on this? The feature in question is disabled by default, maybe they can deprecate it and recommend this extension instead?

**Comment by maptz** [comment]
Ok. This looks really interesting. I'll have a look into this (though it won't be immediate until I get some time).
